app/Repositories/Activity/ActivityRepository.php

```diff
@@ -6,16 +6,190 @@
 use App\Repositories\BaseRepository;
 use App\Repositories\Activity\ActivityRepositoryInterface;
 use Illuminate\Support\Collection;
+use App\Models\Project;
+use App\Models\Playlist;
+use Illuminate\Support\Arr;
+use App\Repositories\H5pElasticsearchField\H5pElasticsearchFieldRepositoryInterface;
+use App\Http\Resources\V1\SearchResource;
+use Illuminate\Database\Eloquent\Builder;
 
 class ActivityRepository extends BaseRepository implements ActivityRepositoryInterface
 {
+    private $h5pElasticsearchFieldRepository;
+
     /**
      * ActivityRepository constructor.
      *
      * @param Activity $model
+     * @param H5pElasticsearchFieldRepositoryInterface $activityRepository
      */
-    public function __construct(Activity $model)
+    public function __construct(Activity $model, H5pElasticsearchFieldRepositoryInterface $h5pElasticsearchFieldRepository)
     {
         parent::__construct($model);
+        $this->h5pElasticsearchFieldRepository = $h5pElasticsearchFieldRepository;
+    }
+
+    /**
+     * Get the search request
+     *
+     * @param  Request  $request
+     * @return Collection
+     */
+    public function searchForm($request)
+    {
+        $projects = [];
+
+        $searchedModels = Activity::searchForm()
```

Comment:
$this->model

Reply:
Replaced "Activity::searchForm()" with "$this->model".